prometheus/promhttp/http.go

```diff
@@ -379,6 +383,9 @@ type HandlerOpts struct {
 	// NOTE: This feature is experimental and not covered by OpenMetrics or Prometheus
 	// exposition format.
 	ProcessStartTime time.Time
+	// If true, WithUnit adds the unit to the encoder options, ultimately allowing the
+	// unit into the final output, provided that EnableOpenMetrics is also true.
+	WithUnit bool
```

Comment:
In https://github.com/prometheus/client_golang/pull/1408, we're changing a bit the structure of the HandlerOpts regarding OpenMetrics functionality. We'll probably want something similar here

Reply:
@ArthurSens I made some changes to my code accordingly, but I think it would be better for your PR to be merged before mine, as yours has already been approved and also because in yours you explain the rationale behind the new HandlerOpts struct and what options to prefer. I can then take it from there and finish mine. What do you think? :)